ios/RCTTWVideoModule.m

```diff
@@ -712,7 +727,26 @@ - (void)remoteDataTrack:(nonnull TVIRemoteDataTrack *)remoteDataTrack didReceive
 
 - (void)remoteDataTrack:(nonnull TVIRemoteDataTrack *)remoteDataTrack didReceiveData:(nonnull NSData *)message {
     // TODO: Handle didReceiveData
-    NSLog(@"DataTrack didReceiveData");
+    NSLog(@"1DataTrack didReceiveData");
+//    NSLog(@"%@", message);
+//    NSLog(@"%@", [message description]);
+
+//    NSString *decodedString=[[NSString alloc] initWithBytes:[message bytes] length:[message length] encoding:NSUTF8StringEncoding];
+//    NSString *decodedString= [NSString stringWithUTF8String:[message bytes]];
+//    NSString* decodedString = [[NSString alloc] initWithData:message encoding: NSASCIIStringEncoding];
+    NSString* decodedString = [[NSString alloc] initWithData:message encoding:NSUTF8StringEncoding];
```

Comment:
Similarly here, you need to skip this step since the data is not a string. I'm very rusty with Objective-C, but something along these lines should work:

```objc
- (void)remoteDataTrack:(nonnull TVIRemoteDataTrack *)remoteDataTrack didReceiveData:(nonnull NSData *)message {
    NSString *encodedBinaryMessage = [message base64EncodedStringWithOptions:0];
    [self sendEventCheckingListenerWithName:dataTrackBinaryMessageReceived body:@{ @"message": encodedBinaryMessage, @"trackSid": remoteDataTrack.sid }];
}
```
```